src/Components/ContextMenu.re

```diff
@@ -0,0 +1,270 @@
+open Oni_Core;
+
+open Revery;
+open Revery.UI;
+open Revery.UI.Components;
+
+module Option = Utility.Option;
+
+module Constants = {
+  let menuWidth = 200;
+  // let maxMenuHeight = 600;
+};
+
+// TYPES
+
+module Id: {
+  type t;
+  let create: unit => t;
+} = {
+  type t = int;
+
+  let lastId = ref(0);
+  let create = () => {
+    incr(lastId);
+    lastId^;
+  };
+};
+
+[@deriving show({with_path: false})]
+type item('data) = {
+  label: string,
+  // icon: option(IconTheme.IconDefinition.t),
+  data: [@opaque] 'data,
+};
+
+type placement = {
```

Comment:
This `placement` is awesome! This will be useful for all kinds of hover / context menu like UI 💯

Reply:
The `originX` and `originY` terms don't seem very intuitive though. I need to find some better terminology for describing positioning relative to a point.